src/tickit_devices/eiger/eiger.py

```diff
@@ -55,142 +72,145 @@ def __init__(
         self.monitor_config: MonitorConfig = MonitorConfig()
         self.monitor_callback_period = SimTime(int(1e9))
 
-    async def initialize(self) -> None:
-        """Function to initialise the Eiger."""
-        self._set_state(State.IDLE)
+        self._num_frames_left: int = 0
+        self._total_frames: int = 0
+        self._data_queue: Queue = Queue()
+        self._series_id: int = 0
 
-    async def arm(self) -> None:
-        """Function to arm the Eiger."""
-        self._set_state(State.READY)
+        self._finished_aquisition: Optional[asyncio.Event] = None
 
-        header_detail = self.stream_config["header_detail"]["value"]
+    @property
+    def finished_aquisition(self) -> asyncio.Event:
+        """Event that is set when an acqusition series is complete.
 
-        json = {
-            "htype": "dheader-1.0",
-            "series": "<id>",
-            "header_detail": header_detail,
-        }
-        if header_detail != "none":
-            config_json = {}
-            disallowed_configs = ["flatfield", "pixelmask" "countrate_correction_table"]
-            for field_ in fields(self.settings):
-                if field_.name not in disallowed_configs:
-                    config_json[field_.name] = vars(self.settings)[field_.name]
+        Property ensures the event is created.
+        """
+        if self._finished_aquisition is None:
+            self._finished_aquisition = asyncio.Event()
 
-        LOGGER.debug(json)
-        LOGGER.debug(config_json)
+        return self._finished_aquisition
 
-    async def disarm(self) -> None:
-        """Function to disarm the Eiger."""
+    async def initialize(self) -> None:
+        """Initialize the detector.
+
+        Required for all subsequent operations.
+        """
         self._set_state(State.IDLE)
 
-        json = {"htype": "dseries_end-1.0", "series": "<id>"}
+    async def arm(self) -> None:
+        """Arm the detector.
 
-        LOGGER.debug(json)
+        Required for triggering.
+        """
+        self._series_id += 1
+        self.stream.begin_series(self.settings, self._series_id)
+        self._num_frames_left = self.settings.nimages
+        self._set_state(State.READY)
 
-    async def trigger(self) -> str:
-        """Function to trigger the Eiger.
+    async def disarm(self) -> None:
+        """Disarm the detector.
 
-        If the detector is in an external trigger mode, this is disabled as
-        this software command interface only works for internal triggers.
+        Intended for use when armed. See state diagram in class docstring.
         """
+        self._set_state(State.IDLE)
```

Comment:
Should we check the current state? What happens if we disarm while acquiring?

Reply:
The real thing will disallow it. This will allow it, it won't actually break though. I was considering making an issue to properly follow the eiger state machine at a later date.